e2e/utils/test-utils.ts

```diff
@@ -267,6 +255,22 @@ export async function findOrCreateWorkspaceCard(
   return cardFound;
 }
 
+/**
+ * Find a suitable workspace among existing workspaces with OWNER role and older than 30 minutes.
+ */
+export async function findAllCards(page: Page): Promise<WorkspaceCard[]> {
+  const existingCards: WorkspaceCard[] = await WorkspaceCard.findAllCards(page, WorkspaceAccessLevel.Owner);
+  // Filter to include Workspaces older than 30 minutes
+  const halfHourMilliSec = 1000 * 60 * 30;
+  const now = Date.now();
+  return Promise.all(
+    await filter(existingCards, async (card) => {
+      const workspaceTime = Date.parse(await card.getLastChangedTime());
+      return now - workspaceTime > halfHourMilliSec;
+    })
```

Comment:
This may make it a bit more concise
```suggestion
  const halfHourAgoMillis =  Date.now() -1000 * 60 * 30;
  return Promise.all(
    await filter(existingCards, async (card) => halfHourAgoMillis > Date.parse(await card.getLastChangedTime());
  )
```

Reply:
👍🏻 I like it.